ow_materials/include/MaterialDistributionPlugin.h

```diff
@@ -0,0 +1,37 @@
+ // The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef MATERIAL_DISTRIBUTION_PLUGIN_H
+#define MATERIAL_DISTRIBUTION_PLUGIN_H
+
+#include <memory>
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/common/common.hh>
+
+#include <AxisAlignedGrid.h>
+#include <MaterialDatabase.h>
+
+namespace ow_materials
+{
+
+class MaterialDistributionPlugin : public gazebo::VisualPlugin
+{
+  public:
+  void Load(gazebo::rendering::VisualPtr parent, sdf::ElementPtr sdf);
+
+  private:
+  gazebo::rendering::VisualPtr m_visual;
+
+  std::unique_ptr<MaterialDatabase> m_material_db;
+
+  std::unique_ptr<AxisAlignedGrid<MaterialID>> m_grid;
```

Comment:
In order to blend between materials, I think this should be a grid of a type other than MaterialID. Maybe something like this:

`class MaterialBlend {
  // How much of each material type.
  // This could take up a lot of space if there were 256 materials, but having the ability to blend materials means we should be able to keep the material count low.
  float m_fractions[num_materials];

  // You might want terramechanics properties like this derived from the blended material that are required during grinding and digging. This would take more space in the data structure but would save computation time if there are frequent queries.
  float m_hardness;

  // We probably don't need a color member because those would be computed once and put into a 3D texture map.

  // Make all fractions sum to 1.0
  void normalize();
};`